capa/features/extractors/ida/insn.py

```diff
@@ -81,11 +83,27 @@ def extract_insn_api_features(fh: FunctionHandle, bbh: BBHandle, ih: InsnHandle)
         # tuple (<module>, <function>, <ordinal>)
         for name in capa.features.extractors.helpers.generate_symbols(api[0], api[1]):
             yield API(name), ih.address
+        return
 
     # check calls to extern functions
     for api in check_for_api_call(insn, get_externs(fh.ctx)):
         # tuple (<module>, <function>, <ordinal>)
         yield API(api[1]), ih.address
+        return
+
+    # extract dynamically resolved APIs stored in renamed globals (renamed for example using `renimp.idc`)
+    if insn.Op1.type == ida_ua.o_mem:
+        op_addr = insn.Op1.addr
+        op_name = idaapi.get_name(op_addr)
+        # check that the memory address has a name and a type
```

Comment:
When giving a global the name of an API IDA known, it gives the function type. To ensure we are not extracting something completely wrong, I am checking that the address has a type. We could remove this check if you think it is not necessary.

I am not checking what type it has, although we expect it to be a function definition, as this complicates the code a bit and we have to ensure it does not break if IDA changes the syntax. But we could check the type using a regular expression that matches strings like:
```
int (__stdcall *WSAStartup)(WORD wVersionRequested, LPWSADATA lpWSAData);
void (__stdcall *Sleep)(DWORD dwMilliseconds);
```

Reply:
all of the above should be in a code comment, so that future readers understand the context and why you wrote the code this way.